### PR TITLE
Game over screen

### PR DESCRIPTION
**Pull request by tomasdivito**
Well, the game over screen is a thing now at least

### CODE REVIEW

File: app/gameover.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  def button_down(button)
+    case button
+    when Gosu::KbReturn then
+        Global.instance.window.state = Menu.new
+    end
+  end
+
+  def draw
+    #@screen.draw 0, 0, GAMEOVER_Z_INDEX
```
Comment (nsatragno) [inline app/gameover.rb]: nit: remove this comment.

File: app/gameover.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @message_font.draw("G A M E  O V E R", ApplicationWindow::WINDOW_WIDTH / 2 - 250, ApplicationWindow::WINDOW_HEIGHT / 2, 1, 1.0, 1.0,
+      Gosu::Color::WHITE)
+    @other_font.draw("Press RETURN to go back to the menu", ApplicationWindow::WINDOW_WIDTH / 2 - 250, (ApplicationWindow::WINDOW_HEIGHT / 2) + 60, 1, 1.0, 1.0,
```
Comment (nsatragno) [inline app/gameover.rb]: nit: could you extract some of these magic numbers into constants?

File: app/player.rb [hunk 3]
```diff
@@ -36,8 +45,21 @@ def collision_points
...
+    if @state != :inmune
+      @lifes -= 1 if @lifes > 0
+      @state = :inmune
+      die! if @lifes == 0
+    end
+  end
+
   def die!
-    @state = :dead
+    #@state = :dead
```
Comment (nsatragno) [inline app/player.rb]: nit: remove this comment.

**Review by nsatragno — COMMENTED** [review]
General comment:
I haven't downloaded and tested the patch but it looks like the game over screen pops up immediately after dying. Could we have a few frames where we display the death animation? Other than that, this LGTM.

File: app/gameover.rb (hunk 2 above)
Comment (tomasdivito) [reply app/gameover.rb]: I think for some reason when i declare a variable with `ApplicationWindow::WINDOW_HEIGHT` i got an error but i will try

Comment (nsatragno) [reply app/gameover.rb]: I mean the 250, 60 and 1.0 that can be found on L21 and L19.

Comment (tomasdivito) [reply app/gameover.rb]: Oh that's ok

**Comment by tomasdivito** [comment]
I fixed the things that you marked on me but the part of the animation, i will introduce it in another patch.

**Review by nsatragno — APPROVED** [review]
Everything looks good, I'll merge this PR once #1 lands.

**Comment by tomasdivito** [comment]
Now you should be able to merge this with no problem :D

**Comment by tomasdivito** [comment]
Wait, no, it's not as it should... mmm

**Comment by nsatragno** [comment]
Looks like the merge got messed up. Ping me if you need help merging (also, it's "Lives", not "Lifes" :))

**Comment by nsatragno** [comment]
Merged as #7

**Pull request closed without merging**